lib/Source.ml

```diff
@@ -116,6 +83,17 @@ let tokens_between (t : t) ~filter loc_start loc_end =
       in
       loop i []
 
+let empty_line_between (t : t) p1 p2 =
+  let l = tokens_between t ~filter:(function _ -> true) p1 p2 in
+  let rec loop prev_lnum l =
+    match l with
+    | [] -> p2.pos_lnum - prev_lnum > 1
+    | (_, x) :: xs ->
+        x.Location.loc_start.pos_lnum - prev_lnum > 1
+        || loop x.loc_end.pos_lnum xs
```

Comment:
If there are tokens in the line between it can still return true with the recursive calls, we need to return false and stop if there are tokens on the line between.
There is a large diff with test_branch.sh with the conventional profile.

Reply:
Isn't it the semantic previously implemented by `&& List.exists (lines_between t p1 p2) ~f:is_line_empty`

Reply:
Yes it is

Reply:
The locations of COMMENTs where wrong, fixed